libcontainer/standard_init_linux.go

```diff
@@ -44,7 +43,22 @@ func (l *linuxStandardInit) getSessionRingParams() (string, uint32, uint32) {
 // the kernel
 const PR_SET_NO_NEW_PRIVS = 0x26
 
-func (l *linuxStandardInit) Init(s chan os.Signal) error {
+func (l *linuxStandardInit) Init() error {
+	var (
+		goStarted   = make(chan struct{}, 1)
+		waitForExec = make(chan struct{}, 1)
+	)
+	// use channel 'c' to ensure that the goroutine was scheduled so that the open is called
+	// before we pivot into the rootfs or else we won't have access to the path on the host.
+	go func() {
+		close(goStarted)
```

Comment:
If there is no race free way to make this work, we could just use either SIGUSR1/SIGUSR2 signals that are reserved for user code.